client/src/components/utilities.ts

```diff
@@ -125,3 +126,31 @@ export const checkAndToast = (
   }
   return false
 }
+
+// https://overreacted.io/making-setinterval-declarative-with-react-hooks/
+/* istanbul ignore next */
+export const useInterval = (
+  callback: Function,
+  delay: number,
+  callImmediately: boolean
+) => {
+  const savedCallback = useRef<Function>()
+
+  // Remember the latest function.
+  useEffect(() => {
+    savedCallback.current = callback
+  }, [callback])
+
+  // Set up the interval.
+  useEffect(/* eslint consistent-return: "off" */
+  () => {
+    function tick() {
+      savedCallback.current!()
+    }
+    if (callImmediately) tick()
+    if (delay !== null) {
```

Comment:
impossible for `delay` to be `null` according to the type

Reply:
Good catch, I'll fix that.

Reply:
@jonahkagan This is fixed by adding `null` to the type. The point of this is so that you can have a state variable controlling the delay, and change it to `null` to pause the loop. It's a feature of the original function that I thought could be useful at some point and so I left it in. The original didn't have types, and I forgot to include that when I added them. So this isn't an error check, it's actually a feature.

Reply:
ah cool!